Docs: Document about roles, people and processes (#1230)
openstreetmap/id-tagging-schema: MAINTENANCE.md

The maintainer role includes:
- Having the final say in decisions for the tagging schema.
- Creating releases.
- Updating dependencies.

and all the following roles.

### Contributor Role

- [Kyle](https://github.com/k-yle) contributes to this project as a volunteer.
- [Tobias](https://github.com/tordans) contributes to this project as a volunteer.

The contributor role includes:
- Reviewing PRs.
- Merging "clear-cut" PRs.

and all the following roles.

### Triage Role

(No one has the triage role at the moment. Please reach out if you want to help.)

The triage role includes:
- Helping to clarify issues and PRs.
- Closing issues as duplicates or not planned.

### Contributors

To all contributors, thank you so much for your support! ❤ Especially for:
- Suggesting new presets and fields or updates to the repository.
- Researching and helping with issues and PRs.
- Translating the tagging schema.

Code contributions: [Check this complete list of contributors on GitHub](https://github.com/openstreetmap/id-tagging-schema/graphs/contributors).

## Processes

### PR Reviews and Merges

> - PRs need approval from two people: the author and one more contributors before being merged.

Reviewer: ```suggestion
- PRs need approval from two people: the author and one or more contributors before being merged.
```

Reviewer: There are other ways to amend this. BTW, is "contributor" requiring any experience/presence? Past accepted PR? It is kind of unclear

Author: > BTW, is "contributor" requiring any experience/presence? Past accepted PR? It is kind of unclear

I just added a very high level section to handle the question how the defined roles are assigned https://github.com/openstreetmap/id-tagging-schema/pull/1230/files#diff-87a0c4623bffa64c60b62a39eeab1c40bd0f329cebc177f5e4492555fb5d42aeR68-R73
This does not go into details on which criteria are used to make this decision which is IMO fine because in the absence of published criteria it is clear IMO that the person responsible for the step will have to make a judgement decision. 

Pink @tyrasd

- Non-"clear-cut" changes need to be merged by the maintainer.
- We might revert merges later if necessary.

**What is a clear-cut change?**

- No or minimal controversial discussion on the change.
- Coding and contribution [guidelines](./GUIDELINES.md) are met.

**How to merge…**

- Usually sqash merge PRs to make the history simpler

> - Give the Merge a meaningfull description of the change

Reviewer: ```suggestion
- Give the merge a meaningful description of the change
```

fix typos

# Roles, People, and Processes for Maintaining the Tagging Schema

This document outlines how this project is maintained.

## Roles & People

### Maintainer Role

[Martin](https://github.com/tyrasd) maintains this project as part of his work on the iD Editor project.

> The maintainer role includes:

Reviewer: Are they also approving people to "Contributor" role?

Author: Good point, added

- Having the final say in decisions for the tagging schema.
- Creating releases.
- Updating dependencies.

and all the following roles.

> ### Contributor Role

Reviewer: Maybe different name may be better? Github is using "contributor" to refer to anyone who got commit accepted and to anyone who opened a PR.

![screen10](https://github.com/openstreetmap/id-tagging-schema/assets/899988/32c000e8-f2b4-44bb-95c7-95db6ebc5f90)

Maybe "Helper"?

Author: Good point. Github also uses "collaborator" …
> <img width="537" alt="image" src="https://github.com/openstreetmap/id-tagging-schema/assets/111561/eb80dbfb-c6e2-47a3-a1db-2772c34a557e">

I think that badge appeared once I had triage roles but I am not sure if the name changed. I remember that the triage role added a name that suggested to readers that I had more of a maintainer role, which is not true.

Unfortunately https://docs.github.com/en/organizations/managing-peoples-access-to-your-organization-with-roles/roles-in-an-organization does not explain which badge is added when.

I think to clarify we should add the github badge name to this section so there is a reference.

We might need to test this out by changing my permission @tyrasd

Author: I added the names of the github labels as I understand them ATM. We will need to look for better names of the chapters/roles.

Reviewer: "Collaborator" is when your account is added to project and has any rights assignable on github (at least from my understanding/memory).

"First-time contributor" is on PR of someone who has their first PR in repo (not sure is it overriding other labels)

"Contributor" - their changes were accepted into repo (not sure about edge cases like their PR merged that contained no commits they made or commits authored/committed by them being accepted without opening PR etc)

Reviewer 2: I changed this role to _"Co-Maintainers"_. I think that better represents the huge added value by these volunteers.

- [Kyle](https://github.com/k-yle) contributes to this project as a volunteer.
- [Tobias](https://github.com/tordans) contributes to this project as a volunteer.

The contributor role includes:
- Reviewing PRs.
- Merging "clear-cut" PRs.

and all the following roles.

### Triage Role

(No one has the triage role at the moment. Please reach out if you want to help.)

The triage role includes:

> - Helping to clarify issues and PRs.

Reviewer: Maybe "Helping to clarify issues and PRs." should be moved altogether to section below?

- Closing issues as duplicates or not planned.

> ### Contributors

Reviewer: Note confusion with "Contributor role"

Reviewer 2: I've renamed the above role to "Co-Maintainers".

- Suggesting new presets and fields or updates to the repository.
- Researching and helping with issues and PRs.
- Translating the tagging schema.

Code contributions: [Check this complete list of contributors on GitHub](https://github.com/openstreetmap/id-tagging-schema/graphs/contributors).

## Processes

### PR Reviews and Merges

- PRs need approval from two people: the author and one or more contributors before being merged.
- Non-"clear-cut" changes need to be merged by the maintainer.
- We might revert merges later if necessary.

**What is a clear-cut change?**

- No or minimal controversial discussion on the change.
- Coding and contribution [guidelines](./GUIDELINES.md) are met.

**How to merge…**

- Usually squash merge PRs to make the history simpler
- Give the merge a meaningful description of the change
- Add labels to the PR to simplify creating the changelog: `new-*`, `add-*`, `bug`

### Releases

There is no set release schedule at the moment, but releases usually occur every other month. After this project is released, the projects that rely on the data need to update and release as well.

### Assigning roles

> - The maintenance role is assigned by the OSMF. Historically the maintainer of the iD Editor also maintained this project.

Reviewer: ```suggestion
- The maintenance role is assigned by the OSMF. So far the maintainer of the iD Editor also maintained this project.
```

"historically" is typically used in cases where it was applying in past but is not any more, and implies such situation. At least as far as I know, I am not a native speaker.

(BTW, is it it "iD editor" or "iD Editor"?)

Author: I see your point. I asked ChatGPT for some wording and used parts of that to change the sentence. I also removed the mention of OSMF which is something that the iD Repo can explain better.

Author: Will resolve this so we can create a new comment on the new line.

# Roles, People, and Processes for Maintaining the Tagging Schema

This document outlines how this project is maintained.

## Roles & People

### Maintainer Role

[Martin](https://github.com/tyrasd) maintains this project as part of his work on the iD Editor project.

The maintainer role includes:
- Having the final say in decisions for the tagging schema.
- Creating releases.
- Updating dependencies.
- Assign roles.

and all the following roles.

Github shows a `(Member)` label next to users with full access to this repository and the organisation.

### Contributor Role

- [Kyle](https://github.com/k-yle) contributes to this project as a volunteer.
- [Tobias](https://github.com/tordans) contributes to this project as a volunteer.

The contributor role includes:
- Reviewing PRs.

> - Merging "clear-cut" PRs.

Reviewer: Is it OK for them to merge own PRs if they seem clear cut for them?

Author: Technically they cannot due to branch protection rules. But lets make it explicit. How about…

```suggestion
- Merging "clear-cut" PRs by others.
```
